### PR TITLE
Added authentication using a query param

### PR DESCRIPTION
**Pull request by paschembri**

### CODE REVIEW

File: concrete_datastore/api/v1/authentication.py [hunk 1]
```diff
@@ -1,6 +1,6 @@
 # coding: utf-8
+import uuid
```
Comment (lcognat) [inline concrete_datastore/api/v1/authentication.py]: not used. Remove that line

File: concrete_datastore/api/v1/authentication.py [hunk 2]
```diff
@@ -95,3 +95,16 @@ def authenticate_credentials(self, key):
             expire_secure_token(secure_token)
 
         return (token.user, token)
+
+
+class URLTokenExpiryAuthentication(TokenExpiryAuthentication):
```
Comment (lcognat) [inline concrete_datastore/api/v1/authentication.py:99]: Please explain within a comment the purpose of that class

File: concrete_datastore/api/v1/authentication.py [hunk 3]
```diff
@@ -95,3 +95,16 @@ def authenticate_credentials(self, key):
...
+    def authenticate(self, request):
+        token = request.GET.get('c_auth_with_token', b'')
+        if token == b'':
+            return
+
+        if len(token) != 40:
```
Comment (lcognat) [inline concrete_datastore/api/v1/authentication.py:105]: what about UID as a token (self registration)

**Review by lcognat — CHANGES_REQUESTED** [review]
that class could be used within settings `DEFAULT_AUTHENTICATION_CLASSES` in `REST_FRAMEWORK` dict

whould be used in 
- EmailDeviceAuthView
- UnBlockUsersApiViewset

File: concrete_datastore/api/v1/authentication.py (hunk 1 above)
Comment (paschembri) [reply concrete_datastore/api/v1/authentication.py]: ok

File: concrete_datastore/api/v1/authentication.py (hunk 2 above)
Comment (paschembri) [reply concrete_datastore/api/v1/authentication.py:99]: docstring added

File: concrete_datastore/api/v1/authentication.py (hunk 3 above)
Comment (paschembri) [reply concrete_datastore/api/v1/authentication.py:110]: Created an issue : https://github.com/Netsach/concrete-datastore/issues/65

**Pull request merged**